toolchain/check/cpp/location.cpp

```diff
@@ -0,0 +1,76 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/check/cpp/location.h"
+
+#include "toolchain/sem_ir/absolute_node_id.h"
+#include "toolchain/sem_ir/ids.h"
+
+namespace Carbon::Check {
+
+static auto GetFile(Context& context, SemIR::CheckIRId ir_id)
+    -> const SemIR::File* {
+  if (ir_id == context.sem_ir().check_ir_id()) {
+    // Common case: the IR is the current file.
+    return &context.sem_ir();
+  }
+
+  // If the file is imported, locate it in our imports map.
+  auto import_id = context.check_ir_map().Get(ir_id);
+  if (!import_id.has_value()) {
+    // We never imported this CheckIR.
+    // TODO: Can this happen?
+    return nullptr;
+  }
```

Comment:
I don't think this should be possible, maybe just CHECK-fail if it does? Perhaps then the caller can assume the return value is non-null?

```suggestion
  CARBON_CHECK(import_id.has_value());
```